### PR TITLE
Add HardDrop functional tests

### PR DESCRIPTION
**Pull request by AndriyLozovyyPLV**
Add HardDrop functional tests:
- test_hw_drop_l3[1]
- test_hw_drop_l3[3]
- test_hw_drop_l3_exp_counters

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/conftest.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    yield
+
+    out = await IpNeighbor.show(input_data=[{dev_name: [{'cmd_options': '-j -4'}]}], parse_output=True)
+    assert not out[0][dev_name]['rc'], 'Failed to get IPv4 neighbors'
+    arps = out[0][dev_name]['parsed_output']
+
+    out = await IpNeighbor.delete(input_data=[{dev_name: [
+        {'dev': arp['dev'], 'address': arp['dst']}
+        for arp in arps if arp['dev'] in dut_ports
+    ]}])
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/conftest.py]: You can multiply a list by a number. This way you don't need to call `IpNeighbor.delete` twice:
```suggestion
    ] * 2}])
```

```python
[1, 2, 3] * 2  # [1, 2, 3, 1, 2, 3]
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    out = out.replace('\x00', '')
+    counters = {}
+    if out:
+        for line in out.splitlines():
+            stat_code, pkts = line.split(':')
+            counters[int(stat_code)] = int(pkts)
+    return counters
+
+
+def verify_expected_counters(counters_before, counters_after):
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]:
```suggestion
def verify_expected_counters(counters_old, counters_new):
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    """
+    Verify that only expected L3 hard_drop counters are increment
+    Args:
+        counters_before (dict): Dict with counters before
+        counters_after (dict): Dict with counters after
+    """
+    for k, v in counters_after.items():
```
Comment (yuriilisovskyi) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]: Change k,v to more specific name

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+    if ports_num == 1:
+        swp_info = {}
+        await tgen_utils_get_swp_info(dent_dev, dut_ports[1], swp_info)
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['ip_source'] = dev_groups[tg_ports[1]][0]['name']
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['ip_destination'] = dev_groups[tg_ports[0]][0]['name']
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['dstMac'] = swp_info['mac']
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['srcMac'] = macs[dut_ports[1]]
+
+    # 5.Transmit traffic and verify drop counters updated according to the rate packets sent
+    for key in DROP_COUNTERS_MAP:
```
Comment (yuriilisovskyi) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]: Can we start all traffic items and verify all drop counters with one traffic send session?

**Review by yuriilisovskyi — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py (hunk 4 above)
Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]: In theory, this should be possible (at least if cpu codes for traffic items do not overlap). @AndriyLozovyyPLV can you please check this?

Comment (AndriyLozovyyPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]: Cpu codes for drop packets do overlap for traffic, that's why I'm sending one stream per time

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py (hunk 3 above)
Comment (AndriyLozovyyPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]: Done

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py (hunk 2 above)
Comment (AndriyLozovyyPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py]: Done

**Pull request closed without merging**